test/Libraries/Microsoft.Extensions.DataIngestion.Tests/Chunkers/SemanticSimilarityChunkerTests.cs

```diff
@@ -0,0 +1,250 @@
+            IReadOnlyList<IngestionChunk<string>> chunks = await chunker.ProcessAsync(doc).ToListAsync();
+            Assert.Single(chunks);
+            Assert.Equal(text, chunks[0].Content);
+        }
+
+        [Fact]
+        public async Task TwoTopicsParagraphs()
+        {
+            IngestionDocument doc = new IngestionDocument("doc");
+            string text1 = ".NET is a free, cross-platform, open-source developer platform for building many" +
+                "kinds of applications. It can run programs written in multiple languages, with C# being the most popular.";
+            string text2 = "It relies on a high-performance runtime that is used in production by many high-scale apps.";
+            string text3 = "Zeus is the chief deity of the Greek pantheon. He is a sky and thunder god in ancient Greek religion and mythology.";
+            doc.Sections.Add(new IngestionDocumentSection
+            {
+                Elements =
+                {
+                    new IngestionDocumentParagraph(text1),
+                    new IngestionDocumentParagraph(text2),
+                    new IngestionDocumentParagraph(text3)
+                }
+            });
+
+            using var customGenerator = new TestEmbeddingGenerator
+            {
+                GenerateAsyncCallback = async (values, options, ct) =>
+                {
+                    var embeddings = values.Select((_, index) =>
+                    {
+                        return index switch
+                        {
+                            0 => new Embedding<float>(new float[] { 1.0f, 1.0f, 1.0f, 1.0f }),
+                            1 => new Embedding<float>(new float[] { 1.0f, 1.0f, 1.0f, 1.0f }),
+                            2 => new Embedding<float>(new float[] { -1.0f, -1.0f, -1.0f, -1.0f }),
+                            _ => throw new InvalidOperationException("Unexpected call count")
+                        };
+                    }).ToArray();
+
+                    return [.. embeddings];
+                }
+            };
+
+            IngestionChunker<string> chunker = CreateSemanticSimilarityChunker(customGenerator);
+            IReadOnlyList<IngestionChunk<string>> chunks = await chunker.ProcessAsync(doc).ToListAsync();
+            Assert.Equal(2, chunks.Count);
+            Assert.Equal(text1 + Environment.NewLine + text2, chunks[0].Content);
+            Assert.Equal(text3, chunks[1].Content);
+        }
+
+        [Fact]
+        public async Task TwoSeparateTopicsWithAllKindsOfElements()
+        {
+            string dotNetTableMarkdown = @"| Language | Type | Status |
+| --- | --- | --- |
+| C# | Object-oriented | Primary |
+| F# | Functional | Official |
+| Visual Basic | Object-oriented | Official |
+| PowerShell | Scripting | Supported |
+| IronPython | Dynamic | Community |
+| IronRuby | Dynamic | Community |
+| Boo | Object-oriented | Community |
+| Nemerle | Functional/OOP | Community |";
+
```

Comment:
nit: Thanks to other reviewers I've learned last week that we can use the `"""` syntax (the closing tag sets indentation level) which is slightly more readable.

```suggestion
            string dotNetTableMarkdown = """
            | Language | Type | Status |
            | --- | --- | --- |
            | C# | Object-oriented | Primary |
            | F# | Functional | Official |
            | Visual Basic | Object-oriented | Official |
            | PowerShell | Scripting | Supported |
            | IronPython | Dynamic | Community |
            | IronRuby | Dynamic | Community |
            | Boo | Object-oriented | Community |
            | Nemerle | Functional/OOP | Community |
            """;

```